src/likelihoods/categorical.jl

```diff
@@ -1,16 +1,20 @@
 """
-    CategoricalLikelihood
+    CategoricalLikelihood(l::AbstractLink=SoftMaxLink())
 
 Categorical likelihood is to be used if we assume that the 
 uncertainity associated with the data follows a Categorical distribution.
 ```math
-    p(y|f_1, f_2, \\dots, f_{n-1}) = Categorical(y | softmax(f_1, f_2, \\dots, f_{n-1}, 0))
+    p(y|f_1, f_2, \\dots, f_{n-1}) = Categorical(y | l(f_1, f_2, \\dots, f_{n-1}, 0))
 ```
-On calling, this would return a Categorical distribution with `f_i` 
-probability of `i` category.
+On calling, this would return a Categorical distribution where probabilities are 
+given by `l(f_1, f_2, ..., f_{n-1}, 0)`, a vector of probabilities, which sums to 1
```

Comment:
I wonder whether this should be a little bit more literal, and make it clear that you need to provide a vector, not a tuple, as input?